File: src/main/java/org/jukeboxmc/event/player/PlayerRespawnEvent.java
package org.jukeboxmc.event.player;

import org.jukeboxmc.math.Location;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class PlayerRespawnEvent extends PlayerEvent {

    private Location respawnLocation;

    /**
     * Creates a new {@link PlayerEvent}
     *
     * @param player who represents the player which comes with this event
     */
    public PlayerRespawnEvent( Player player, Location respawnLocation ) {
        super( player );
        this.respawnLocation = respawnLocation;
    }

    public Location getRespawnLocation() {
        return this.respawnLocation;
    }

    public void setRespawnLocation( Location respawnLocation ) {
        this.respawnLocation = respawnLocation;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/player/PlayerRespawnEvent.kt
package org.jukeboxmc.api.event.player

import org.jukeboxmc.api.math.Location
import org.jukeboxmc.api.player.Player

class PlayerRespawnEvent(
    player: Player,
    private var respawnLocation: Location
) : PlayerEvent(player) {

    fun getRespawnLocation(): Location {
        return this.respawnLocation
    }

    fun setRespawnLocation(respawnLocation: Location) {
        this.respawnLocation = respawnLocation
    }

}